src/WebJobs.Script/Diagnostics/IMetricsLogger.cs

```diff
@@ -12,6 +12,13 @@ public interface IMetricsLogger
         /// <summary>
         /// Begins an event.
         /// </summary>
+        /// <param name="eventName">The event name.</param>
+        /// <returns>A handle to the event that was started.</returns>
+        object BeginEvent(string eventName);
+
+        /// <summary>
+        /// Begins an event.
+        /// </summary>
         /// <param name="metricEvent">The event.</param>
         void BeginEvent(MetricEvent metricEvent);
```

Comment:
I'll probably look into whether we can unify these APIs further. For these system events, I wanted to be able to instrument the code in a lightweight fashion, passing in event names as strings.  However, some of the existing events are passing in event subclasses that have additional data, and don't conform to this model.